api/src/main/java/bio/terra/tanagra/service/search/Filter.java

```diff
@@ -74,5 +76,66 @@ public static ArrayFunction create(List<Filter> operands, Operator operator) {
     }
   }
 
-  // TODO implement a relationship filter tht allows chaining to another entity.
+  /**
+   * A {@link Filter} that introduces a constraint that there must be a relationship to another
+   * entity.
+   *
+   * <p>This quantifiers introduces a new {@link Variable} and implicitly binds it to an entity in
+   * {@link #boundAttribute()}.
+   *
+   * <p>This is equivalent to the pseudo-SQL ":outerAttribute IN (SELECT :boundAttribute FROM
+   * :boundAttribute.entity AS :boundAttribute.variable WHERE :filter".
+   *
+   * <p>Or in words ∃x∈boundEntity( x.boundAttribute = outerAttribute ∧ filter): there exists {@link
+   * #boundAttribute()} that's a member of {@link #boundAttribute()}'s Entity where {@link
```

Comment:
I'm having a bit of difficulty following this, so I may be wrong, but should the first term on this line be `#boundEntity()` instead?